linkerd/app/src/env.rs

```diff
@@ -929,11 +944,29 @@ fn parse_addr(s: &str) -> Result<Addr, ParseError> {
     })
 }
 
-fn parse_port_set(s: &str) -> Result<HashSet<u16>, ParseError> {
-    let mut set = HashSet::new();
+fn parse_port_range_set(s: &str) -> Result<RangeInclusiveSet<u16>, ParseError> {
+    let mut set = RangeInclusiveSet::new();
     if !s.is_empty() {
-        for num in s.split(',') {
-            set.insert(parse_number::<u16>(num)?);
+        for part in s.split(',') {
+            let mut parts = part.splitn(2, '-');
+            let low = parts.next().ok_or_else(|| {
+                error!("Not a valid port range: {part}");
+                ParseError::NotAPortRange
+            })?;
+            let low = parse_number::<u16>(low)?;
+            if let Some(high) = parts.next() {
+                let high = parse_number::<u16>(high).map_err(|e| {
+                    error!("Not a valid port range: {part}");
+                    e
+                })?;
+                if high < low {
+                    error!("Not a valid port range: {part}; {high} is greater than {low}");
+                    return Err(ParseError::NotAPortRange);
+                }
+                set.insert(low..=high);
+            } else {
+                set.insert(low..=low);
+            }
```

Comment:
It's probably appropriate to add some tests for this, now that it's getting a bit more complicated. There's a `mod tests` at the bottom of the file.

Reply:
Will do!